python/cudf/dataframe/dataframe.py

```diff
@@ -1766,9 +1766,10 @@ def quantile(self, q, interpolation='linear', exact=False):
         result = DataFrame()
         result['Quantile'] = q
```

Comment:
I recommend 

```python
if columns is None:
    columns = self.columns
```

Reply:
I like this and will do it

Reply:
Following up to your earlier point for the very lightweight pdf<->gdf test (nice!), does `assert_eq` work with floating point error?

Reply:
Yes, `assert_eq` passes keyword arguments down to the appropriate pandas testing function

https://github.com/rapidsai/cudf/blob/b74591b723c79a1c052642ec356879fe688ae0d4/python/cudf/tests/utils.py#L36-L62

So we can check those docs here: https://pandas.pydata.org/pandas-docs/stable/generated/pandas.testing.assert_frame_equal.html

And learn that we probably want to use `check_exact=False`

Reply:
I've implemented your requested test as an xfail:

It turns out that pdf.quantile() returns a Series. gdf.quantile() returns a DataFrame. gdf.quantile() cannot be refactored to return a Series at this time, because string Series indexes are not supported yet in cudf.